File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/users/inmemory/UserRepositoryInMemory.java
package io.zenwave360.example.clinicaltool.modules.users.inmemory;

import io.zenwave360.example.clinicaltool.modules.users.domain.*;
import io.zenwave360.example.clinicaltool.modules.users.UserRepository;
import java.math.*;
import java.time.*;
import java.util.*;

public class UserRepositoryInMemory extends InMemoryJpaRepository<User> implements UserRepository {
    @Override
    public java.util.Optional<User> findByUsername(String username) {
        return getEntities().values().stream().filter(e ->
             isSameValue(username, readField(e, "username")) 
        ).findFirst();
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/users/inmemory/UserRepositoryInMemory.kt
package io.zenwave360.example.clinicaltool.modules.users.inmemory

import io.zenwave360.example.clinicaltool.modules.users.UserRepository
import io.zenwave360.example.clinicaltool.modules.users.domain.*
import java.math.*
import java.time.*
import java.util.*

class UserRepositoryInMemory : InMemoryJpaRepository<User, Long>(), UserRepository {
    override fun findByUsername(username: String): User? {
        return getEntities().values.firstOrNull { e -> isSameValue(username, readField(e, "username")) }
    }
}
